src/main.cpp

```diff
@@ -3856,8 +3856,9 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                     }
                 }
                 // disconnect node in case we have reached the outbound limit for serving historical blocks
+                // never disconnect whitelisted nodes
```

Comment:
I think it's more or less adequately explained by the documentation for whitelist.

Though the release note for maxupload target could mention it too.